homeassistant/components/omnilogic/sensor.py

```diff
@@ -0,0 +1,484 @@
+                "device_class": None,
+                "icon": "mdi:gauge",
+                "unit": None,
+                "guard_condition": {
+                    "Shared-Type": "BOW_SHARED_EQUIPMENT",
+                    "status": "0",
+                },
+            },
+        ],
+        (6, "CSAD"): [
+            {
+                "entity_classes": {"ph": OmniLogicPHSensor},
+                "name": "pH",
+                "kind": "csad_ph",
+                "device_class": None,
+                "icon": "mdi:gauge",
+                "unit": None,
+                "guard_condition": {"ph": ""},
+            },
+            {
+                "entity_classes": {"orp": OmniLogicORPSensor},
+                "name": "ORP",
+                "kind": "csad_orp",
+                "device_class": None,
+                "icon": "mdi:gauge",
+                "unit": None,
+                "guard_condition": {"orp": ""},
+            },
+        ],
+    }
+
+    coordinator = hass.data[DOMAIN][entry.entry_id][COORDINATOR]
+    entities = []
+
+    for item_id, item in coordinator.data.items():
+        id_len = len(item_id)
+        item_kind = item_id[-2]
+        entity_settings = sensor_types.get((id_len, item_kind))
+
+        if not entity_settings:
+            continue
+
+        for entity_setting in entity_settings:
+            for state_key, entity_class in entity_setting["entity_classes"].items():
+                if state_key not in item:
+                    continue
+
+                guard_condition = entity_setting["guard_condition"]
+                if guard_condition and all(
+                    item.get(guard_key) == guard_value
+                    for guard_key, guard_value in guard_condition.items()
+                ):
+                    continue
+
+                entity = entity_class(
+                    coordinator,
+                    state_key,
+                    entity_setting["name"],
+                    entity_setting["kind"],
+                    item_id,
+                    entity_setting["device_class"],
+                    entity_setting["icon"],
+                    entity_setting["unit"],
+                )
+
+                entities.append(entity)
+
+    async_add_entities(entities, update_before_add=True)
+
+
+class OmnilogicSensor(OmniLogicEntity):
+    """Defines an Omnilogic sensor entity."""
+
+    def __init__(
+        self,
+        coordinator: OmniLogicUpdateCoordinator,
+        kind: str,
+        name: str,
+        device_class: str,
+        icon: str,
+        unit: str,
+        item_id: tuple,
+        state_key: str,
+    ):
+        """Initialize Entities."""
+        super().__init__(
+            coordinator=coordinator,
+            kind=kind,
+            name=name,
+            item_id=item_id,
+            icon=icon,
+        )
+
+        backyard_id = item_id[:2]
+        unit_type = coordinator.data[backyard_id].get("Unit-of-Measurement")
+
+        self._state = None
```

Comment:
We don't need the state instance attribute since we always calculate state in the state property.

Reply:
Addressed